Tests/TestingTests/MiscellaneousTests.swift

```diff
@@ -552,17 +552,6 @@ struct MiscellaneousTests {
     #expect(id.keyPathRepresentation == [""])
   }
 
-  @Test("Test.all deduping")
```

Comment:
Why did you remove this test? Did it start failing? Can it be refactored?

Reply:
This PR removes `Test.testsByID()` and modifies `Test.all` to directly return all runtime-enumerated test instances, so I didn't see a remaining need for this test.

Reply:
We still want to test that content is deduped because the raw content we collect can have duplicates.

Reply:
Ok, I think that would be better to test at the end of the planning process, whereas this test is validating an intermediary stage (after runtime discovery but before planning). I'll refactor it.

Reply:
Deduping can be a natural consequence of us building the initial test graph, FWIW.

Reply:
Oh yeah, and I recall now that's actually why I didn't bother replacing or refactoring this test when I removed it. I think we're OK as-is